### PR TITLE
Properly type event.data

### PR DESCRIPTION
**Pull request by cramforce**
…to ensure the data is considered an external `JsonObject`. Forces the use of a `getData` function instead of accessing the `.data` property directly. The function returns the proper type.

Part of #9876

### CODE REVIEW

File: build-system/conformance-config.textproto [hunk 1]
```diff
@@ -168,6 +168,22 @@ requirement: {
 }
 
 requirement: {
+  type: BANNED_PROPERTY_READ
```
Comment (aghassemi) [inline build-system/conformance-config.textproto]: Do we need something similar for `event.detail` ?

File: extensions/amp-ad/0.1/amp-ad-xorigin-iframe-handler.js [hunk 2]
```diff
@@ -136,7 +136,7 @@ export class AmpAdXOriginIframeHandler {
     // iframe.
     listenForOncePromise(this.iframe, 'entity-id', true)
         .then(info => {
-          this.element_.creativeId = info.data.id;
+          this.element_.creativeId = info.data['id'];
```
Comment (aghassemi) [inline extensions/amp-ad/0.1/amp-ad-xorigin-iframe-handler.js]: `getData(info)['id']` ? info == event

File: build-system/conformance-config.textproto (hunk 1 above)
Comment (cramforce) [inline build-system/conformance-config.textproto]: No, why that one in particular?
In general we need it when the property contains externally defineable (non-standard) complex data structures.

File: extensions/amp-ad/0.1/amp-ad-xorigin-iframe-handler.js (hunk 2 above)
Comment (cramforce) [inline extensions/amp-ad/0.1/amp-ad-xorigin-iframe-handler.js]: No. Slightly confusingly `listenForOncePromise` returns an internal data structure that looks similar to a browser event, but isn't. It is also properly typed and so enforces the `[]` subscript automatically.

File: build-system/conformance-config.textproto (hunk 1 above)
Comment (aghassemi) [inline build-system/conformance-config.textproto]: `action-impl` adds `detail` property to some event objects. I don't believe `detail` is standard prop of event object.

Comment (choumx) [inline build-system/conformance-config.textproto]: It is a property on `CustomElement` though. We should probably change action-impl's implementation instead, e.g. copy properties into a new `CustomElement` object like we do for `DeferredEvent`.

Comment (cramforce) [inline build-system/conformance-config.textproto]: I see. That is a bit of an ambiguous case. You only strictly need to taint such an object if it cross compilation units. Not really related to this PR.

**Comment by cramforce** [comment]
Ping for approval. Please approve large refactorings quickly to avoid churn from merge conflicts.

**Pull request merged**